### DISCUSSION TITLE
Two or more instances of sqlpage running

### DISCUSSION THREAD

**Discussion by plagonell** [sqlpage/SQLPage#307] [Q&A]
Hi, I did a test using  postgresql with sqlpage and it works fine, but I need to have another instance open for MySql, when I try to open it, it takes the value of the Postgres instance that is running from the .env variable. How to open one or more instances of sqlpage, I am not using docker. Excuse me, English is not my language.

**Comment by lovasoa** [comment]
Hello and welcome to SQLPage!

It sounds like you're trying to use multiple instances of `sqlpage`, one for PostgreSQL and another for MySQL, but the second instance is using the environment variables from the first instance. 

To fix this, you would typically have to specify different environment variables for each instance. If you are running directly from the command line, you can specify environment variables before the executable name in the terminal:

```bash
DATABASE_URL='mysql://...' PORT=8080 ./sqlpage.bin;
DATABASE_URL='postgres://...' PORT=8081 ./sqlpage.bin
```

If you are not using docker, you'll probably want to use something else to manage your services, restart them after a reboot or a crash, etc... Maybe you are using systemd, in which case you can specify environment variables in the [service definition file](https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html)

You can also use configuration files instead of environment variables and use two different configuration files for the two instances: https://github.com/lovasoa/SQLpage/blob/main/configuration.md

**Comment by lovasoa** [comment]
To clarify, are you trying to serve two instances of the same site, connected to two different databases? Or are you creating two different sites? 

It's hard to write sql queries that work the same on both MySQL and postgres.

**Comment by plagonell** [comment]
Hello, the first answer is the one I will take, and it actually worked for
me. Thanks for your prompt response. Again, excellent product. I'll be
watching in the github chat.

El mié, 8 may 2024 a las 16:49, Ophir LOJKINE ***@***.***>)
escribió:

> To clarify, are you trying to serve two instances of the same site,
> connected to two different databases? Or are you creating two different
> sites?
>
> It's hard to write sql queries that work the same on both MySQL and
> postgres.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lovasoa/SQLpage/discussions/307#discussioncomment-9360056>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADCEJWXF6GC6KOHJJ34V6WLZBKFVVAVCNFSM6AAAAABHNN4HXCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TGNRQGA2TM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
.::.
Saludos Cordiales y Prevención, cuídese...

Pedro Lagonell

Tlf: 0416-6154171
